internal/monitors/metricsfilter.go

```diff
@@ -0,0 +1,153 @@
+package monitors
+
+// Filter of datapoints based on included status and user configuration of
+// extraMetrics and extraGroups.
+
+import (
+	"errors"
+	"fmt"
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/signalfx-agent/internal/utils/filter"
+	"strings"
+)
+
+type metricsFilter struct {
+	metadata     *Metadata
+	extraMetrics map[string]bool
+	stringFilter *filter.BasicStringFilter
+}
+
+func validateMetricName(metadata *Metadata, metricName string) error {
+	if strings.TrimSpace(metricName) == "" {
+		return errors.New("metricName cannot be empty")
+	}
+
+	if !metadata.MetricsExhaustive {
+		// The metrics list isn't exhaustive so can't do extra validation.
+		return nil
+	}
+
+	if strings.ContainsRune(metricName, '*') {
+		// Make sure a metric pattern matches at least one metric.
+		m, err := filter.NewBasicStringFilter([]string{metricName})
+		if err != nil {
+			return err
+		}
+
+		for metric := range metadata.Metrics {
```

Comment:
Would this cover the metrics that have parmaters in their name in the metadata (e.g. the `kubernetes.node_<condition_type>` one from `kubernetes-cluster`)?

Reply:
Yeah it allows it through because it's not mark as exhaustive. Maybe we can add the snake case as a special case in the future to allow kubernetes-cluster to be marked as exhaustive.